### PR TITLE
Local Tracing

### PR DESCRIPTION
**Pull request by steve-keep**
Show OpenTelemetry Traces inside the Admin UI:

Screenshot 2024-07-02 at 4 31 30 pm

### CODE REVIEW

File: src/examples/sqlite/databases/database.sqlite
Comment (steve-keep) [inline src/examples/sqlite/databases/database.sqlite:1]: I wonder if this and the trace db should be removed from the examples and instructions added to generate them or a script that does it first time?

File: src/packages/core/src/metadata.ts [hunk 1]
```diff
@@ -30,6 +32,10 @@ export interface EntityMetadata<G = unknown, D = unknown> {
 		// by fields. This is useful for entities that are not backed by a fully fledged provider or are
 		// otherwise dynamically generated.
 		excludeFromFiltering?: boolean;
+
+		// This means that there will be no recorded telemetry for this entity. This is useful for entities
```
Comment (thekevinbrown) [inline src/packages/core/src/metadata.ts:36]:
```suggestion
		// This means that there will be no recorded telemetry for this entity when it's requested from the Admin UI. This is useful for entities
```

File: src/packages/core/src/metadata.ts [hunk 2]
```diff
@@ -30,6 +32,10 @@ export interface EntityMetadata<G = unknown, D = unknown> {
...
+		// that are managed by some other system or we don't want to record telemetry for.
```
Comment (thekevinbrown) [inline src/packages/core/src/metadata.ts:37]:
```suggestion
		// that are managed by some other system or we don't want to record telemetry for.
		//
		// Note that in order to achieve this from your own frontends, you need to set the `X-GRAPHWEAVER-SUPPRESS-TRACING: true` header on the requests, which is what this option instructs the admin UI to do for you.
```

File: src/packages/core/src/open-telemetry/entity.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+	);
+	Field(() => GraphQLNanoDuration, { adminUIOptions: { hideInFilterBar: true } })(
+		Trace.prototype,
+		'duration'
+	);
+	Field(() => String)(Trace.prototype, 'traceId');
+	Field(() => GraphQLJSON, { adminUIOptions: { hideInFilterBar: true, hideInTable: true } })(
+		Trace.prototype,
+		'attributes'
+	);
```
Comment (thekevinbrown) [inline src/packages/core/src/open-telemetry/entity.ts:34]: Could we just factory the whole entity here so our decorators look normal, or no?

**Review by thekevinbrown — APPROVED** [review]

File: src/packages/admin-ui-components/src/entity-list/styles.module.css [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+	opacity: 0.6;
+}
+
+.successToast {
+	display: flex;
+	flex-direction: column;
+}
+
+.deletedText {
+	color: #ede8f2b2;
```
[the remark this answers is not in the document]
Comment (steve-keep) [reply src/packages/admin-ui-components/src/entity-list/styles.module.css:72]: Not currently used anywhere else.

File: src/packages/core/src/open-telemetry/entity.ts (hunk 3 above)
Comment (steve-keep) [reply src/packages/core/src/open-telemetry/entity.ts:34]: Yeah the reason for doing it this way is that the the developer can just apply ACL to the class as normal:

```
import { Trace } from '@exogee/graphweaver';
ApplyAccessControlList({
	DARK_SIDE: {
		// Dark side user role can perform operations on any tag
		all: true,
	},
})(Trace);
```

So it looks odd in this file but better from the end user.

**Pull request merged**